2018/readme.md

```diff
@@ -0,0 +1,128 @@
+# 2018 File Generation Features
+
+## The FileGenerator Class 
+The scripts used in this repository for generating clean files, edit files, and edit reports use methods from the FileGenerator class. Clean files will pass syntax and validity edits. Edit test files will fail at least one (and sometimes more than one) edit. These files will be used as a double check for the implementation of the HMDA edits for the 2018 HMDA Platform. Creating edit test files is a two step process. The generation of edit test files requires a clean data file to be present. If edit files already exist with the same names as those produced by this script, they will be overwritten.
+
+The `FileGenerator()` class is located in [`file_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/file_generator.py) and can be instantiated without any inputs. The class uses data stored with yaml configuration files in the [`python/configurations/`](https://github.com/cfpb/hmda-test-files/tree/master/2018/python/configurations) directory. 
+
+Edit test files are written to the directory indicated in `test_filepaths.yaml`based on edit type, such as /edits_files/test_files/Bank1/syntax/s300.txt. The file name indicates which edit the file is designed to test. If an edit has multiple conditions, a file will be made for each condition in the format /edits_files/test_files/Bank1/syntax/s301_1.txt. If multiple permutations of the same condtion are tested, the files will have an alphabetic naming convention, such as /edits_files/test_files/Bank1/validity/v715_a.txt.
+
+The FileGenerator Class relies on four other classes, each with its own role in file creation:
+
+lar_generator in [`lar_generator.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/lar_generator.py) 
+
+Creates raw TS and LAR rows using the [TS and LAR schema.json files located in /schemas/](https://github.com/cfpb/hmda-test-files/blob/master/2018/schemas) , the values of which are then manipulated to create clean test files or edit test files.
+
+`rules_engine` in [`rules_engine.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/rules_engine.py) 
+
+Contains logic for each edit. The rules engine can generate an edit report for a passed file. This report contains the following fields:
+
+    * edit name
```

Comment:
Is there a reason you used asterisks instead of dashes for these bullets? If not, please change to dashes for consistency.

Reply:
I think this was a typo for adding bullets.